src/gwt/src/org/rstudio/studio/client/workbench/views/output/find/model/FindResult.java

```diff
@@ -58,51 +101,206 @@ public native final String getLineValue() /*-{
       return getJavaArray("matchOff");
    }
 
+   public final ArrayList<Integer> getReplaceMatchOns()
+   {
+      return getJavaArray("replaceMatchOn");
+   }
+
+   public final ArrayList<Integer> getReplaceMatchOffs()
+   {
+      return getJavaArray("replaceMatchOff");
+   }
+
+   public final String getErrors()
+   {
+      return getJavaStringArray("errors");
+   }
+
+   public final native void setReplace(String value) /*-{
+      if (value)
+         this.replace = value;
+      else
+         this.replace = "";
+      this.regexPreviewIndicator = false;
+   }-*/;
+
+
    public final SafeHtml getLineHTML()
    {
       SafeHtmlBuilder out = new SafeHtmlBuilder();
 
-      ArrayList<Integer> on = getMatchOns();
-      ArrayList<Integer> off = getMatchOffs();
-      ArrayList<Pair<Boolean, Integer>> parts
-                                      = new ArrayList<Pair<Boolean, Integer>>();
-      while (on.size() + off.size() > 0)
+      if (getRegexPreviewIndicator() &&
+          !StringUtil.isNullOrEmpty(getErrors()))
       {
-         int onVal = on.size() == 0 ? Integer.MAX_VALUE : on.get(0);
-         int offVal = off.size() == 0 ? Integer.MAX_VALUE : off.get(0);
-         if (onVal <= offVal)
-            parts.add(new Pair<Boolean, Integer>(true, on.remove(0)));
-         else
-            parts.add(new Pair<Boolean, Integer>(false, off.remove(0)));
+         out.appendHtmlConstant("<mark>");
+         out.appendEscapedLines(getErrors());
+         out.appendHtmlConstant("</mark>");
       }
-
-      String line = getLineValue();
-
-      // Use a counter to ensure tags are balanced.
-      int openTags = 0;
-
-      for (int i = 0; i < line.length(); i++)
+      else
       {
-         while (parts.size() > 0 && parts.get(0).second == i)
+         ArrayList<Integer> on = getMatchOns();
+         ArrayList<Integer> off = getMatchOffs();
+         ArrayList<Pair<Boolean, Integer>> parts
+                                         = new ArrayList<Pair<Boolean, Integer>>();
+   
+         ArrayList<Integer> replaceOn = getReplaceMatchOns();
+         ArrayList<Integer> replaceOff = getReplaceMatchOffs();
+         ArrayList<Pair<Boolean, Integer>> replaceParts
+                                      = new ArrayList<Pair<Boolean, Integer>>();
+   
+         int difference = 0;
+         while (on.size() + off.size() > 0)
          {
-            if (parts.remove(0).first)
+            int onVal = on.size() == 0 ? Integer.MAX_VALUE : on.get(0);
+            int offVal = off.size() == 0 ? Integer.MAX_VALUE : off.get(0);
+            int replaceOnVal = replaceOn.size() == 0 ? Integer.MAX_VALUE : replaceOn.get(0);
+            int replaceOffVal = replaceOff.size() == 0 ? Integer.MAX_VALUE : replaceOff.get(0);
+   
+            if (onVal <= offVal)
```

Comment:
If `onVal == offVal` is there anything to do?

Reply:
onVal could == offVal if a prior match range is ending at the same point a following match is beginning. In this scenario, offVal would be added as an off part and on the next iteration of the loop, onVal would be less than offVal (the end of this following match) and would be added as an on part.